cfpb/capital-framework
# Update cf-icons to use inline svgs rather than an icon font

Icon fonts have some shortcomings with accessibility and
maintainability. To simplify usage, we've converted them to inline SVGs
that are then included by each project's templating system. The process
from hand-off from Graphic Designers requires removing left/right
whitespace (we batch processed the icons through
an Illustrator script to save hours of work), followed by optimization
via SVGO within Capital Framework. The resulting icons are then copied
to their respective projects via that project's build process. Finally,
we updated each of the other components that require icons to use the
new pattern. The exception is the example for the rotating icon within
a button (see the usage doc for further info).

## Additions

- SVG Icons
- CSS to size, position and color icons

## Changes

- Converts other CF Components to use the new SVG Icons

## Testing

- Follow the [CONTRIBUTING.md instructions for testing locally](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-components-locally). The site specific branch is `svg-icons-test`
- In `svg-icons-test` run gulp build && npm start
- Navigate to `http://localhost:3000/components/cf-icons/` and read the documentation and review the tables of icons
- Navigate through the rest of the components and ensure icons are working as expected where they are currently used (compare with the live docs for reference).

## Screenshots

![screen shot 2018-02-06 at 2 01 51 pm](https://user-images.githubusercontent.com/1280430/35881405-4ccaf616-0b46-11e8-884a-73b9dac93000.png)

![screen shot 2018-02-06 at 2 02 22 pm](https://user-images.githubusercontent.com/1280430/35881428-5e936d10-0b46-11e8-9f5a-3f6cfe307e19.png)

![screen shot 2018-02-06 at 2 02 45 pm](https://user-images.githubusercontent.com/1280430/35881446-6a9f3a3a-0b46-11e8-8928-18fa32343ff7.png)

![screen shot 2018-02-06 at 2 03 01 pm](https://user-images.githubusercontent.com/1280430/35881462-759c3d7a-0b46-11e8-80a6-cffccef1a65a.png)

## Notes

- There are 400 svg files (200 base file + 200 aliases), you can safely ignore them when reviewing the changed files.
- Due to the fact that non-icon components would stop working after being updated to use the new SVG icon patterns within each of them if a user didn't update cf-icons as well. Scott and I are proposing that we merge this PR to a pre-release branch, and publish that as a pre-release on NPM. We'd then update cfgov-refresh to use the pre-release to give us a month or so to ensure we've worked out any issues.

## Todos

- Update the necessary icon source files (update.svg) to include `-spin` modifiers with the animation code embedded in the SVG.

## Checklist

* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG) SEE NOTES SECTION

## Testing checklist

### Browsers


### Accessibility


### Other
